packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -59,26 +62,30 @@ export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLE
     return class extends HTMLElement {
         constructor() {
             super();
-
-            if (this.isConnected) {
-                // this if block is hit when there's already an un-upgraded element in the DOM with the same tag name.
-                hydrateComponent(this, Ctor, {});
-                hydratedCustomElements.add(this);
-            } else {
-                createVM(this, Ctor, renderer, {
-                    mode: 'open',
-                    owner: null,
-                    tagName: this.tagName,
-                });
+            if (!isNull(this.shadowRoot)) {
+                if (process.env.NODE_ENV !== 'production') {
+                    // eslint-disable-next-line no-console
+                    console.warn(
+                        `Found an existing shadow root for the custom element ${Ctor.name}. Call \`hydrateComponent\` instead.`
+                    );
+                }
+                clearNode(this.shadowRoot);
             }
+            if (this.childNodes.length > 0) {
+                if (process.env.NODE_ENV !== 'production') {
+                    // eslint-disable-next-line no-console
+                    console.warn(`Top level elements cannot have children.`);
```

Comment:
```suggestion
                    console.warn(`Custom elements cannot have child nodes. Ensure the element is empty, including whitespace.`);
```

This might be more helpful, since even whitespace will cause this warning to appear.